Anima/math-tools/special_functions/animaKummerFunctions.cxx

```diff
@@ -88,31 +96,98 @@ KummerMethod2(const double &x,
     return resVal;
 }
 
+double KummerIntegrandMethod(const double &x, const double &a, const double &b)
+{
+    KummerIntegrand integrand;
+    integrand.SetXValue(x);
+    integrand.SetAValue(a);
+    integrand.SetBValue(b);
+    double resVal = boost::math::quadrature::gauss<double, 15>::integrate(integrand, 0.0, 1.0);
+    resVal *= std::tgamma(b) / std::tgamma(a) / std::tgamma(b - a);
+}
+
 double
-KummerFunction(const double &x,
-               const double &a,
-               const double &b,
-               const bool scaled,
-               const bool normalized,
-               const unsigned int maxIter,
-               const double tol)
+GetKummerFunctionValue(const double &x, const double &a, const double &b,
+                       const unsigned int maxIter, const double tol)
 {
-    if (a > 0 && b > 0)
+    if ((a == 0)|| (x == 0))
+        return 1.0;
+
+    if (a == b)
+        return std::exp(x);
+
+    if (a > 0 && b > a)
     {
-        KummerIntegrand integrand;
-        integrand.SetXValue(x);
-        integrand.SetAValue(a);
-        integrand.SetBValue(b);
-        double resVal = boost::math::quadrature::gauss<double, 15>::integrate(integrand, 0.0, 1.0) / std::tgamma(b - a);
-        if (!normalized)
-            resVal *= (std::tgamma(b) / std::tgamma(a));
-        return (scaled || x <= 0) ? resVal : std::exp(x) * resVal;
+        double resVal = KummerIntegrandMethod(x,a,b);
+
+        if (x > 0)
+            resVal = std::exp(x + std::log(resVal));
+
+        return resVal;
     }
-    
-    if (std::abs(x) < 50.0)
+
+    double rFactor = std::abs(x * a / b);
+    if (rFactor < 20.0)
         return KummerMethod1(x,a,b,maxIter,tol);
     else
+    {
+        if (a > b)
+        {
+            double ap = b - a;
+            double xp = -x;
+            if (ap > b)
+                throw itk::ExceptionObject(__FILE__, __LINE__,"Invalid inputs for Kummer function using method 2",ITK_LOCATION);
+
+            double tmpVal = KummerMethod2(xp,ap,b,maxIter,tol);
+            double logResVal = x + std::log(std::abs(tmpVal));
+            double factor = (0 < tmpVal) - (0 > tmpVal);
```

Comment:
Are you sure `(0 > tmpVal)` will be cast as double so that `- (0 > tmpVal)` is well defined ? Wouldn't it be safer to use `double factor = (double)(0 < tmpVal)-(double)(0 > tmpVal)` ?

Reply:
Changing this to compare to 0.0, but yes it will be converted well